apps/system/js/bluetooth_v2.js

```diff
@@ -435,6 +439,49 @@ Bluetooth.prototype = {
   },
 
   /**
+   * Get adapter from bluetooth through promise interface.
+   * XXX: the function abstract the Bluetooth API difference.
+   * We can remove it and use service query once BTv1 is deprecated.
+   *
+   * @public
+   * @return {Promise} Bluetooth Adapter
+   */
+  adapter: function bt__adapter() {
+    return new Promise((resolve, reject) => {
+      if (this._adapter !== null) {
+        resolve(this._adapter);
+      } else {
+        this.debug('No BT adapter retrieved');
+        reject();
+      }
+    });
+  },
+
+  /**
+   * Return device pairing result.
+   * XXX: the function abstract the Bluetooth API difference.
+   *
+   * @public
+   * @param {string} mac target device address
+   * @return {Promise} paired result
+   */
+  pair: function bt__pair(mac) {
+    return this._adapter.pair(mac);
+  },
+
+  /**
+   * Return paired devices list.
+   * XXX: the function abstract the Bluetooth API difference.
+   * We can use service query to return list once BTv1 is deprecated.
+   *
+   * @public
+   * @returns {Object[]} sequence of BluetoothDevice
+   */
+  getPairedDevices: function bt__getPairedDevices() {
+    return this._adapter.getPairedDevices();
```

Comment:
Service.request will wrap promise interface for us, so we don't have to do it by our self
https://github.com/mozilla-b2g/gaia/blob/master/apps/system/js/service.js#L57